Client key for read operations

smallhive opened the pull request:
closes #429

smallhive commented:
Many commits, to be able to make changes in each place separately. Before merging, we may squash them for few

cthulhu-rider reviewed (commented):
overall LGTM. Lets sync minor documentation moments

cthulhu-rider commented on client/container.go:
```diff
@@ -24,15 +24,6 @@ type PrmContainerPut struct {
-}
-
-// SetSigner sets signer to sign request payload.
-// Signer's scheme MUST be neofscrypto.ECDSA_DETERMINISTIC_SHA256. For example, you can use neofsecdsa.SignerRFC6979.
```
we must not loose this requirement in method docs

cthulhu-rider replied:
@smallhive good, but one note: this restriction applies only to container operations now (as an additional benefit, this allows us to exec object ops with different schemes)

smallhive replied:
Got it, removed from object operations

cthulhu-rider commented on client/client.go:
```diff
@@ -50,19 +52,18 @@ type Client struct {
+		return nil, fmt.Errorf("private key: %w", err)
 	}
+
+	prm.signer = neofsecdsa.SignerRFC6979(pk.PrivateKey)
```
with this we can get rid of nil-check in methods like `NetworkInfo`, right? They will never return missing signing error now

smallhive replied:
Removed Signer check inside these methods

smallhive commented on client/object_delete.go:
```diff
@@ -102,14 +87,17 @@ func (x *PrmObjectDelete) WithXHeaders(hs ...string) {
 //
 // Context is required and must not be nil. It is used for network communication.
 //
+// Signer is required and must not be nil. It is used to sign request data. The wallet of this account will be charged
```
Sound better, updated

carpawell reviewed (commented):
1. Conflicts.
2. There is a commit about the tests, does that mean that before that commit the tests did not pass? I would not go that way: I think after every commit build and tests should be OK.
3. The last commit does not end with a dot and also it seems more like "Created/Is created" rather than "Creates".

smallhive commented:
> 1. Conflicts.
> 2. There is a commit about the tests, does that mean that before that commit the tests did not pass? I would not go that way: I think after every commit build and tests should be OK.
> 3. The last commit message does not end with a dot and also it seems more like "Created/Is created" rather than "Creates".

1. Fixed
2. I agree. Meanwhile, these tests are marked by a special build tag and don't affect test pipelines. But ok, divided commit and moved to another
3. What do you mean? I don't use dots in commits ends. And didn't catch an idea about meaning :smiley:

cthulhu-rider commented:
> I think after every commit build and tests should be OK

this makes sense, but sometimes you just add unit test which shows some bug (so test fails), and then add fix commit (which in particular fixes test) so u can feel the impact of new changes. Otherwise, test should always pass, yeah

cthulhu-rider commented:
@smallhive i guess last commit's message looks a bit wierd and out-of-context, we could cleary write
`Touched methods never return ErrMissingSigner because signer is initialized in the constructor, so nil-check is always passed.`

but i don't insist, me personally can realize any form

smallhive commented:
> @smallhive i guess last commit's message looks a bit wierd and out-of-context, we could clearly write `Touched methods never return ErrMissingSigner because signer is initialized in the constructor, so nil-check is always passed.`
> 
> but i don't insist, me personally can realize any form

I just wanted to left some context in extended commit message

cthulhu-rider commented:
> I just wanted to left some context in extended commit message

and this is great and must have, but iiuc @carpawell was confused by grammar

carpawell commented:
> Meanwhile, these tests are marked by a special build tag and don't affect test pipelines.

> but sometimes you just add unit test which shows some bug (so test fails), and then add fix commit (which in particular fixes test) so u can feel the impact of new changes.

Well, I would not go that way anyway. As it is described in our [git](https://github.com/nspcc-dev/.github/blob/master/git.md) page with rules, "This is important to have an ability to bisect problematic changes if there is a need to."

> I don't use dots in commits ends.

Oh, well, I was not attentive enough then. Hm, it is not declared explicitly but we have an example [there](https://github.com/nspcc-dev/.github/blob/master/git.md):

> subsystem: short description (up to 80 symbols, better up to 60)
>
> Long description with references, links, test data and whatever else is needed to explain what, why and how is changed.
> 
> Signed-off-by: Your Name [email@example.com](mailto:email@example.com)

I would interpret that as a rule that says that any sentence should end with a period (if it is not a header). Anyway, I would say that all the commits we have done until that PR end with dots if they contain a body.

> And didn't catch an idea about meaning

I meant that I did not get the idea of the "Creates in the constructor". Who creates? What are that words about?

smallhive commented:
> @smallhive, I meant that I did not get the idea of the "Creates in the constructor". Who creates? What are that words about?

I got it, I updated the description messages in the commits
